File: Samples/Silverlight_RiaServices_Hierarchical_VB/Silverlight_RiaServices.Web/Validators/Orders.Validation.vb
'===============================================================================
'                   EntitySpaces Studio by EntitySpaces, LLC
'            Persistence Layer and Business Objects for Microsoft .NET
'            EntitySpaces(TM) is a legal trademark of EntitySpaces, LLC
'                         http://www.entityspaces.net
'===============================================================================
' EntitySpaces Version : 2012.1.0930.0
' EntitySpaces Driver  : SQL
' Date Generated       : 9/23/2012 6:16:25 PM
'===============================================================================

Imports System
Imports System.ComponentModel.DataAnnotations

Namespace BusinessObjects

	<MetadataType(GetType(OrdersValidation))> _
	Public Partial Class Orders

	End Class

    Friend Class OrdersValidation
    

		<KeyAttribute()> _
		<Editable(false)> _
		Public Property OrderID As Nullable(Of System.Int32)

		<StringLength(5)> _
		Public Property CustomerID As System.String

		Public Property EmployeeID As Nullable(Of System.Int32)

		Public Property OrderDate As Nullable(Of System.DateTime)

		Public Property RequiredDate As Nullable(Of System.DateTime)

		Public Property ShippedDate As Nullable(Of System.DateTime)

		Public Property ShipVia As Nullable(Of System.Int32)

		Public Property Freight As Nullable(Of System.Decimal)

		<StringLength(40)> _
		Public Property ShipName As System.String

		<StringLength(60)> _
		Public Property ShipAddress As System.String

		<StringLength(15)> _
		Public Property ShipCity As System.String

		<StringLength(15)> _
		Public Property ShipRegion As System.String

		<StringLength(10)> _
		Public Property ShipPostalCode As System.String

		<StringLength(15)> _
		Public Property ShipCountry As System.String
    
	End Class
End Namespace

File: Samples/Silverlight_RiaServices_Hierarchical_CS/Silverlight_RiaServices.Web/Validators/Orders.Validation.cs
/*===============================================================================
                    EntitySpaces Studio by EntitySpaces, LLC
             Persistence Layer and Business Objects for Microsoft .NET
             EntitySpaces(TM) is a legal trademark of EntitySpaces, LLC
                          http://www.entityspaces.net
===============================================================================
EntitySpaces Version : 2012.1.0930.0
EntitySpaces Driver  : SQL
Date Generated       : 9/23/2012 6:16:17 PM
===============================================================================
*/

using System;
using System.ComponentModel.DataAnnotations;

namespace BusinessObjects
{
    [MetadataType(typeof(OrdersValidation))]
    public partial class Orders
    {

    }
	
    internal class OrdersValidation
    {

		[KeyAttribute()]
		[Editable(false)]
		public System.Int32? OrderID { get; set; }

		[StringLength(5)]
		public System.String CustomerID { get; set; }

		public System.Int32? EmployeeID { get; set; }

		public System.DateTime? OrderDate { get; set; }

		public System.DateTime? RequiredDate { get; set; }

		public System.DateTime? ShippedDate { get; set; }

		public System.Int32? ShipVia { get; set; }

		public System.Decimal? Freight { get; set; }

		[StringLength(40)]
		public System.String ShipName { get; set; }

		[StringLength(60)]
		public System.String ShipAddress { get; set; }

		[StringLength(15)]
		public System.String ShipCity { get; set; }

		[StringLength(15)]
		public System.String ShipRegion { get; set; }

		[StringLength(10)]
		public System.String ShipPostalCode { get; set; }

		[StringLength(15)]
		public System.String ShipCountry { get; set; }
    }
}